test-lib.sh

```diff
@@ -229,8 +228,8 @@ function ct_assert_container_creation_fails() {
 function ct_create_container() {
   local cid_file="$CID_FILE_DIR/$1" ; shift
   # create container with a cidfile in a directory for cleanup
-  # shellcheck disable=SC2086,SC2153
-  docker run --cidfile="$cid_file" -d ${CONTAINER_ARGS:-} "$IMAGE_NAME" "$@"
+  # shellcheck disable=SC2153
+  docker run --cidfile="$cid_file" -d "${CONTAINER_ARGS:-}" "$IMAGE_NAME" "$@"
```

Comment:
This will not work - you are passing cli arguments, we want them to be handled separately, not as a single string.

Reply:
For and idea on how to fix this, you can take a look at: https://github.com/sclorg/container-common-scripts/pull/201/files#diff-dd412cc49cd894778cdf696290d87106f2f66e59c3887bec15e68c64b27bf21eL760-R761

Reply:
Good idea, what I have found in the our CONTAINER_ARGS usage:
E.g. httpd-container https://github.com/sclorg/httpd-container/blob/master/2.4/test/run#L69
```
CONTAINER_ARGS="-e HTTPD_LOG_TO_VOLUME=1 --user 0 -v ${logs_dir}:${volume_dir}"
```

E.g. mysql-container and similar mariadb-container https://github.com/sclorg/mysql-container/blob/master/test/run#L538
```
CONTAINER_ARGS="run-mysqld --innodb_buffer_pool_size=5242880"
```

Correct would be:
For httpd-container results are:
```bash
CONTAINER_ARGS="-e HTTPD_LOG_TO_VOLUME=1 --user 0 -v ${logs_dir}:${volume_dir}"
```
```bash
$ read -d'\n' -ra cont_args <<< "$(echo "$CONTAINER_ARGS" | grep -o -e '\.*\S*' || true)"
$ echo ${cont_args[@]+"${cont_args[@]}"}
HTTPD_LOG_TO_VOLUME=1 --user 0 -v /tmp/logs:/var/logs
$
```
why -e disappear, I don't know

For mariadb-container results are:
```bash
CONTAINER_ARGS="run-mysqld --innodb_buffer_pool_size=5242880"
```
```bash
$ read -d'\n' -ra cont_args <<< "$(echo "$CONTAINER_ARGS" | grep -o -e '\.*\S*' || true)"
$ echo ${cont_args[@]+"${cont_args[@]}"}
run-mysqld --innodb_buffer_pool_size=5242880
$
```

Reply:
But I will update PR.

Reply:
Is there a reason we are filling the array like this? What use is there for the grep?

For the missing `-e` - pretty sure `echo` is eating it as it believes it is passed as an option for its call. Do we need to use `echo` in the first place?

Reply:
We can directly use `<<< $CONTAINER_ARGS`, but I did not try it.

Reply:
I expect `cont_args=("$CONTAINER_ARGS")` would work as well

Reply:
Useful, but not resolving problem with -e, see:
```bash
$ CONTAINER_ARGS="-e HTTPD_LOG_TO_VOLUME=1 --user 0 -v /tmp/dir1:/tmp/dir2"
$ cont_args=($CONTAINER_ARGS)
$ echo ${cont_args[@]+"${cont_args[@]}"}
HTTPD_LOG_TO_VOLUME=1 --user 0 -v /tmp/dir1:/tmp/dir2
$

Reply:
Tested on RHEL7, but when I move `-e` option at the end all works fine.

Reply:
Of course, your code snippet still has `echo` for checking the array contents. If you print it out with `printf` then you will see the `-e` is in the array.